plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/databinding/ProjectIdValidator.java

```diff
@@ -0,0 +1,34 @@
+package com.google.cloud.tools.eclipse.ui.util.databinding;
+
+import java.util.regex.Pattern;
+
+import org.eclipse.core.databinding.validation.IValidator;
+import org.eclipse.core.databinding.validation.ValidationStatus;
+import org.eclipse.core.runtime.IStatus;
+
+import com.google.cloud.tools.eclipse.ui.util.Messages;
+
+public class ProjectIdValidator implements IValidator {
+  // todo: this is different from .newproject.AppEngineProjectIdValidator
```

Comment:
we're going to want to get rid of that class after this is in. 
What is the difference between that and this?